lib/iris/_merge.py

```diff
@@ -407,11 +407,6 @@ def match(self, other, error_on_mismatch):
         if self.data_type != other.data_type:
             msg = 'cube data dtype differs: {} != {}'
             msgs.append(msg.format(self.data_type, other.data_type))
-        if self.fill_value != other.fill_value:
-            # Allow fill-value promotion if either fill-value is None.
-            if self.fill_value is not None and other.fill_value is not None:
-                msg = 'cube fill value differs: {} != {}'
-                msgs.append(msg.format(self.fill_value, other.fill_value))
```

Comment:
@lbdreyer This is more in alignment with the behaviour of `upstream/master`, which doesn't discriminate based on `fill_value`. So no breaking or surprising behaviour for the end user.

It also aligns with the behaviour of `numpy`, which will merge/concatenate arrays with different `fill_values`, but the resultant array has a `fill_value` that is replaced with the `numpy` nominated default for that `dtype`. 

With regards to cubes, we're doing the same - if we have two cubes that only differ on `fill_value`, say one is `1234` and the other is `4321`, then they will merge/concatenate into one cube, but the resultant cube will have a `fill_value` of `None. When the user does a `cube.data` to get the data, the resulting masked array (assuming the payload realizes is masked) will have a `fill_value` that is the `numpy` nominated default.